gguf-py/scripts/gguf_template.py

```diff
@@ -0,0 +1,159 @@
+"""
+gguf_template.py - example file to extract the chat template from the models metadata
+"""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import sys
+from pathlib import Path
+
+import jinja2
```

Comment:
This is a dependency of `transformers`, so it should already be in the user's environment, but it might be relevant to add it to the dev-dependencies in the `pyproject.toml` of the `gguf-py` library.

[`transformers` seems to use](https://github.com/huggingface/transformers/blob/a5c642fe7a1f25d3bdcd76991443ba6ff7ee34b2/setup.py#L127) version `>=3.1.0` of `jinja2`.

Reply:
If that's alright with @ggerganov. Yeah, I could add it explicitly. I just operated under the implicit dependency for transformers library.

Reply:
Yes, it's ok to add it explicitly as dev dependency

Reply:
Are the `pyproject.toml` intended to be compatible with `poetry`?

Some kind of weird issue with `sentencepiece` versus when I use the `requirements.txt` where it can't find a shared object dependency.

```sh
ImportError: libprotobuf-lite.so.25: cannot open shared object file: No such file or directory
```

Didn't notice until I attempted to execute the script. `requirements.txt` works like it always did. No issues there.

Reply:
> Are the `pyproject.toml` intended to be compatible with `poetry`?

cc @ditsuke @SomeoneSerge (this is related to <https://github.com/ggerganov/llama.cpp/pull/5745>)

Reply:
Yeah, upgrading to `sentencepiece-0.2.0` from `sentencepiece-0.1.99` fixes the issue for me.

Reply:
> ImportError: libprotobuf-lite.so.25: cannot open shared object file: No such file or directory

Seems like an issue with the prebuild packages on PyPi.

> Are the pyproject.toml intended to be compatible with poetry?

`pyproject.toml` currently uses `poetry` as the build back-end; at a glance it seems to work fine:

```
❯ mkdir _build
❯ cd _build
❯ pip wheel ..
❯ python -m pip install llama_cpp_scripts-0.0.0-py3-none-any.whl
...
  Using cached gguf-0.9.1-py3-none-any.whl
...
```

Reply:
So is this constraint thing blocking the PR right now? I think it's OK for you to sneak in a commit updating the constraint and re-generating the lock file

Reply:
No, just making sure there's consensus.

If it's not a big deal, I can bump `sentencepiece` to `0.2.0` in the `pyproject.toml`.

```toml
sentencepiece = ">=0.2.0"
```

Reply:
Yes, I think it's by mistake that `pyproject.toml` diverged from `requirements.txt`. Update the constraint and run `poetry lock` (maybe `poetry lock --no-update` to minimize the diff; @ditsuke would know better I guess) please

Reply:
It should be fine to bump in pyproject, then `poetry update sentencepiece` to update the pin in the lockfile.